packages/nc-gui-v2/components/smartsheet-column/CurrencyOptions.vue

```diff
@@ -0,0 +1,102 @@
+<script setup lang="ts">
+import { useColumnCreateStoreOrThrow, useProject } from '#imports'
+import { currencyCodes, currencyLocales, validateCurrencyCode, validateCurrencyLocale } from '@/utils'
+
+interface Option {
+  label: string
+  value: string
+}
+
+const { formState, validateInfos, setAdditionalValidations, sqlUi, onDataTypeChange, onAlter } = useColumnCreateStoreOrThrow()
+
+const validators = computed(() => {
+  return {
+    'meta.currency_locale': [
+      {
+        validator: (_: any, locale: any) => {
+          return new Promise<void>((resolve, reject) => {
+            if (!validateCurrencyLocale(locale)) {
+              return reject(new Error('Invalid locale'))
+            }
+            resolve()
+          })
+        },
+      },
+    ],
+    'meta.currency_code': [
+      {
+        validator: (_: any, currencyCode: any) => {
+          return new Promise<void>((resolve, reject) => {
+            if (!validateCurrencyCode(currencyCode)) {
+              return reject(new Error('Invalid Currency Code'))
+            }
+            resolve()
+          })
+        },
+      },
+    ],
+  }
+})
+
+// set additional validations
+setAdditionalValidations({
+  ...validators.value,
+})
+
+const { isPg } = useProject()
+
+const currencyList = ref(currencyCodes)
+
+const currencyLocaleList = ref(currencyLocales())
```

Comment:
Another unnecessary ref

Reply:
Also add `|| []` so it can be avoided in the `for ... of ...` loop in v-for down below.